datalad/distribution/dataset.py

```diff
@@ -384,11 +393,14 @@ def get_superdataset(self, topmost=False):
             # None was found
             return None
 
-        if realpath(self.path) != self.path:
+        # and it is a subdirectory, not some fancy symlink somewhere outside
+        if realpath(self.path) != self.path and \
+            realpath(self.path).startswith(with_pathsep(realpath(sds_path))):
```

Comment:
FTR:  in https://github.com/datalad/datalad/pull/1011/files#diff-3947667888c8ff9d4bfc6c26f2be70f9L387 I overhauled this a bit -- apparently I didn't need this check here when `normpath` is used.